src/legacy/ui/public/agg_types/controls/extended_bounds.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useEffect } from 'react';
+
+import { EuiFieldNumber, EuiFlexGroup, EuiFlexItem, EuiFormRow, EuiIconTip } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { AggParamEditorProps } from 'ui/vis/editors/default';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+interface Bounds {
+  min: string;
+  max: string;
+}
+
+function isBoundEmpty(value: string): boolean {
+  return value === '';
+}
+
+function areBoundsInvalid({ min, max }: Bounds): boolean {
+  if (isBoundEmpty(min) && isBoundEmpty(max)) {
+    return false;
+  }
+
+  const minValue: number = parseInt(min, 10);
+  const maxValue: number = parseInt(max, 10);
+
+  if (isNaN(minValue) || isNaN(maxValue)) {
+    return true;
+  }
+
+  return minValue > maxValue;
+}
+
+function ExtendedBoundsParamEditor({ value, setValue, setValidity }: AggParamEditorProps<Bounds>) {
+  const tooltipContent = i18n.translate('common.ui.aggTypes.extendedBoundsTooltip', {
+    defaultMessage:
+      'Min and Max do not filter the results, but rather extend the bounds of the result set',
+  });
+
+  const mainLabel = (
+    <>
+      <FormattedMessage
+        id="common.ui.aggTypes.extendedBoundsLabel"
+        defaultMessage="Extended bounds"
```

Comment:
Should this say `(optional)` as it did in the past?

Reply:
Actually instead of saying "optional" let's make it a switch that is always visible but it's `disable` prop is tied to the state of the "Show empty buckets". Like so:

<img src="https://d.pr/free/i/fEBjRv+" width="50%" />

<img src="https://d.pr/free/i/p1qNSy+" width="50%" />

<img src="https://d.pr/free/i/19Us5X+" width="50%" />

Reply:
Then you can also remove the EuiIconTip and place the tooltip directly on the switch as the "Show empty buckets" switch does.

Reply:
> Should this say `(optional)` as it did in the past?

It was not optional previously:
![image](https://user-images.githubusercontent.com/31325372/56720340-50f5b980-674b-11e9-8b2f-03f88d3d6c2f.png)

Optional were `Min` and `Max` values which is wrong. Available options are described in the PR description: 
- neither `Max` nor `Min` bound are set - `VALID`;
- both `Max` and `Min` are set - `VALID` if `Max` >= Min`;
- only one bound set - `INVALID`;
- Min > Max - `INVALID`;

Reply:
Discussed with @maryia-lapata , adding an additional switch button would overload the sidebar (in our race to make it lightweight). Maybe adding the `(optional)` text would make sense: 

![image](https://user-images.githubusercontent.com/31325372/56730375-8953c280-6760-11e9-907b-e0497b93d205.png)

Reply:
So you're first bullet about the validation:

> neither Max nor Min bound are set - VALID;

Does essentially mean that adding values to the Extended bounds is optional. The confusing part is adding the words (optional) but needing both values to be applied if one is. So by using a switch the user is essentially declaring "I want to add values" and so you can then check to make sure their input is valid. I liked the switch because it actually declutters the UI when a user doesn't want add those values. Instead they're staring at 2 empty fields they'll never touch.